hawkbit-rest-resource/src/test/java/org/eclipse/hawkbit/rest/resource/TargetResourceTest.java

```diff
@@ -920,6 +1026,8 @@ private String generateStatusreferenceLink(final String knownTargetId, final Act
         final List<Target> updatedTargets = deploymentManagement.assignDistributionSet(one, targets)
                 .getAssignedTargets();
         // 2nd update
+        // sleep 1ms to ensure that we can sort by reportedAt
+        Thread.sleep(1);
```

Comment:
This sleep could also be removed by using the `AuditingHandler`

By setting an `DateTimeProvider` on the `AuditingHandler` for this test and switch back to the old one again
```
  @Autowired
    private AuditingHandler auditingHandler;
    // set for this test a new date time provider
    auditingHandler.setDateTimeProvider(myDateTimeProvider);
   // switch back the old one again
   auditingHandler.setDateTimeProvider(CurrentDateTimeProvider.INSTANCE);
```
I guess a 1ms sleep is sufficient just  worry that maybe a 1ms is not enough when using `System.currentMillis` is not updated every millisecond maybe.
```
public static long currentTimeMillis()
Returns the current time in milliseconds. Note that while the unit of time of the return value is a millisecond, the granularity of the value depends on the underlying operating system and may be larger. For example, many operating systems measure time in units of tens of milliseconds
```